src/com/ra4king/circuitsim/simulator/components/memory/RAM.java

```diff
@@ -104,8 +115,9 @@ public void valueChanged(CircuitState state, WireValue value, int portIndex) {
 				}
 				break;
 			case PORT_CLK:
-				if(!load && value.getBit(0) == State.ONE && address.isValidValue()) {
-					WireValue lastReceived = state.getLastReceived(getPort(PORT_DATA));
+				if(((!isSeparateLoadStore && !load) || (isSeparateLoadStore && store)) && value.getBit(0) == State.ONE && address.isValidValue()) {
+					int data_port_num = (isSeparateLoadStore) ? PORT_DATA_IN : PORT_DATA;
```

Comment:
Plz no snake case, and just put the conditional directly inside getPort(...)

Reply:
lol mb I like snakes